### PR TITLE
Support CUDA 11.4

### PR DESCRIPTION
**Pull request by takagi**

### CODE REVIEW

File: docker.py [hunk 1]
```diff
@@ -963,6 +979,14 @@ def get_scipy_choices():
     include_dir='/usr/include',
     lib_dir='/usr/lib/x86_64-linux-gnu',
 )
+codes['nccl2.9-cuda114'] = nccl_base.format(
+    nccl_sub_dir='cuda',
+    nccl_os_ver='1804',
+    libnccl2='libnccl2_2.9.9-1+cuda11.3_amd64',
+    libnccl_dev='libnccl-dev_2.9.9-1+cuda11.3_amd64',
```
Comment (takagi) [inline docker.py:986]: Seems that NCCL provides a binary for each minor version, and the one for CUDA 11.4 is not put yet. Skip `'nccl2.9-cuda114'` this time?

File: run_test.py [hunk 2]
```diff
@@ -304,11 +304,11 @@ def main():
 
         conf = {
             'base': 'ubuntu18_py36',
-            'cuda': 'cuda113',
-            'cudnn': 'cudnn82-cuda113',
-            'nccl': 'nccl2.9-cuda113',
+            'cuda': 'cuda114',
```
Comment (takagi) [inline run_test.py:307]: Any preference in configuration to test CUDA 11.4?

**Comment by takagi** [comment]
pfnCI, test this please.

File: docker.py (hunk 1 above)
Comment (kmaehashi) [reply docker.py:986]: I think it's ok to do this for now.

Comment (kmaehashi) [reply docker.py:986]: Raised an issue for record #675

File: run_test.py (hunk 2 above)
Comment (kmaehashi) [reply run_test.py:307]: The configuration LGTM 👍🏼

**Comment by kmaehashi** [comment]
Please wait until our cluster team bumps the CUDA driver on Jenkins node.

This test failure is not related to CUDA 11.4 but seems new, looking into it.

```
17:44:25 =================================== FAILURES ===================================
17:44:25 ___________________ TestNewDLPackConversion.test_conversion ____________________
17:44:25 
17:44:25 self = <cupy_tests.core_tests.test_dlpack.TestNewDLPackConversion testMethod=test_conversion>
17:44:25 dtype = <class 'numpy.uint8'>
17:44:25 
17:44:25     @testing.for_all_dtypes(no_bool=True)
17:44:25     def test_conversion(self, dtype):
17:44:25         allowed_streams = ['null', True]
17:44:25         if not cuda.runtime.is_hip:
17:44:25             allowed_streams.append('ptds')
17:44:25     
17:44:25         # stream order is automatically established via DLPack protocol
17:44:25         for src in allowed_streams:
17:44:25             src_s = self._get_stream(src)
17:44:25             for dst in allowed_streams:
17:44:25                 dst_s = self._get_stream(dst)
17:44:25                 with src_s:
17:44:25                     orig_array = _gen_array(dtype)
17:44:25                 with dst_s:
17:44:25                     out_array = cupy.from_dlpack(orig_array)
17:44:25 >                   testing.assert_array_equal(orig_array, out_array)
17:44:25 
17:44:25 allowed_streams = ['null', True, 'ptds']
17:44:25 dst        = True
17:44:25 dst_s      = <Stream 151227216 (device 0)>
17:44:25 dtype      = <class 'numpy.uint8'>
17:44:25 orig_array = array([[6, 9, 6],
17:44:25        [1, 7, 4]], dtype=uint8)
17:44:25 out_array  = array([[6, 9, 6],
17:44:25        [1, 7, 4]], dtype=uint8)
17:44:25 self       = <cupy_tests.core_tests.test_dlpack.TestNewDLPackConversion testMethod=test_conversion>
17:44:25 src        = 'ptds'
17:44:25 src_s      = <Stream 2 (device -1)>
17:44:25 
17:44:25 cupy_tests/core_tests/test_dlpack.py:64: 
17:44:25 _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
17:44:25 
17:44:25 x = array([[6, 9, 6],
17:44:25        [1, 7, 4]], dtype=uint8)
17:44:25 y = array([[6, 9, 6],
17:44:25        [1, 7, 4]], dtype=uint8), err_msg = ''
17:44:25 verbose = True, strides_check = False
17:44:25 
17:44:25     def assert_array_equal(x, y, err_msg='', verbose=True, strides_check=False):
17:44:25         """Raises an AssertionError if two array_like objects are not equal.
17:44:25     
17:44:25         Args:
17:44:25              x(numpy.ndarray or cupy.ndarray): The actual object to check.
17:44:25              y(numpy.ndarray or cupy.ndarray): The desired, expected object.
17:44:25              strides_check(bool): If ``True``, consistency of strides is also
17:44:25                  checked.
17:44:25              err_msg(str): The error message to be printed in case of failure.
17:44:25              verbose(bool): If ``True``, the conflicting values
17:44:25                  are appended to the error message.
17:44:25     
17:44:25         .. seealso:: :func:`numpy.testing.assert_array_equal`
17:44:25         """
17:44:25         numpy.testing.assert_array_equal(
17:44:25             cupy.asnumpy(x), cupy.asnumpy(y), err_msg=err_msg,
17:44:25 >           verbose=verbose)
17:44:25 E       AssertionError: 
17:44:25 E       Arrays are not equal
17:44:25 E       
17:44:25 E       Mismatched elements: 6 / 6 (100%)
17:44:25 E       Max absolute difference: 255
17:44:25 E       Max relative difference: 255.
17:44:25 E        x: array([[0, 0, 0],
17:44:25 E              [0, 0, 0]], dtype=uint8)
17:44:25 E        y: array([[6, 9, 6],
17:44:25 E              [1, 7, 4]], dtype=uint8)
17:44:25 
17:44:25 err_msg    = ''
17:44:25 strides_check = False
17:44:25 verbose    = True
17:44:25 x          = array([[6, 9, 6],
17:44:25        [1, 7, 4]], dtype=uint8)
17:44:25 y          = array([[6, 9, 6],
17:44:25        [1, 7, 4]], dtype=uint8)
17:44:25 
17:44:25 /home/user/.local/lib/python3.7/site-packages/cupy/testing/_array.py:93: AssertionError
17:44:25 ----------------------------- Captured stdout call -----------------------------
17:44:25 dtype is <class 'numpy.uint8'>
```

**Comment by kmaehashi** [comment]
We noticed that Jenkins worker is Ubuntu 16.04 and CUDA 11.4 dropped support for it.
Instead of maintaining on-premise Jenkins worker, I think it's better to start migrating to FlexCI.
https://github.com/cupy/cupy/pull/5457

**Comment by kmaehashi** [comment]
I've introduced CUDA 11.4 test on FlexCI.
https://github.com/cupy/cupy/pull/5457

**Pull request closed without merging**